test/e2e/framework/kubelet/config.go

```diff
@@ -37,8 +38,8 @@ import (
 )
 
 // GetCurrentKubeletConfig fetches the current Kubelet Config for the given node
-func GetCurrentKubeletConfig(ctx context.Context, nodeName, namespace string, useProxy bool) (*kubeletconfig.KubeletConfiguration, error) {
-	resp := pollConfigz(ctx, 5*time.Minute, 5*time.Second, nodeName, namespace, useProxy)
+func GetCurrentKubeletConfig(ctx context.Context, nodeName, namespace string, useProxy bool, standaloneMode bool) (*kubeletconfig.KubeletConfiguration, error) {
```

Comment:
not for now, but now that I see , I think we should move these things under test/e2e_node, don't we?
@pohly wdyt?
After learning more about e2e_node and see how different is from e2e I think that having they decoupled is the best solution

Reply:
`GetCurrentKubeletConfig` may be handy for e2e as well. But likely not as needed

Reply:
e2e should be completely agnostic  and not interact with any component, it has to express behavior and simulate user experience,

Reply:
I agree with @aojea: test/e2e/framework is for code which is of general use for E2E tests that run as client of an apiserver.

It looks like GetCurrentKubeletConfig fits one aspect ("client of an apiserver") but not the other ("general use") because it is used only by code under `test/e2e_node`. It should be moved there.

The main practical advantage is that code ownership becomes clearer. SIG Node owns test/e2e_node and then can make changes to tests and helper code without having to involve folks from SIG Testing.

`GetCurrentKubeletConfig` isn't too bad because other users of the framework can ignore it. What absolutely should be fixed is that config options and thus command line flags for options that are relevant only to test/e2e_node appear in all E2E suites built on top of the framework.

Reply:
I wonder if we can get some of the e2e_node tests and run them against the cluster. I think it should be possible. Like container lifecycle tests - we can run them in e2e_node for speed, but there is nothing in the test that requires it to be a single node test and those ideally can be made into Conformance.

If we can make them built for both - it may be beneficial. And this may be a good reason to share the framework.

(not specifically a comment to this function though)

Reply:
That sounds like a worthwhile goal.